pgjdbc/src/test/java/org/postgresql/jdbc/TypeInfoCacheGetArrayDelimiterTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import static org.junit.Assert.assertEquals;
+
+import org.postgresql.core.TypeInfo;
+import org.postgresql.jdbc.TypeInfoCacheTestUtil.PgTypeSet;
+import org.postgresql.jdbc.TypeInfoCacheTestUtil.PgTypeStruct;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PSQLException;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class TypeInfoCacheGetArrayDelimiterTest {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  private static PgTypeSet typeSet;
+
+  private Connection conn;
+  private TypeInfo typeInfo;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    Connection conn = TestUtil.openDB();
+    typeSet = PgTypeSet.createAndInstall(types, conn);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    Connection conn = TestUtil.openDB();
+    typeSet.uninstall(conn);
+    TestUtil.closeDB(conn);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    conn = TestUtil.openDB();
+    typeInfo = ((PgConnection) conn).getTypeInfo();
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    TestUtil.closeDB(conn);
+  }
+
+  private final PgTypeStruct type;
+  private final PgTypeStruct arrayType;
+  private final Character delimiter;
+
+  private static final Character DEFAULT_DELIMITER = ',';
+
+  private static final ArrayList<PgTypeStruct> types = new ArrayList<PgTypeStruct>() {
+    {
+      add(new PgTypeStruct("ns", "type"));
```

Comment:
There's likely a lot of issues like this as I haven't coded a lot of Java in the past. I appreciate you pointing them out and ask that you also provide suggestions as to what you'd like to see.